drivers/temperature/adt7420/adt7420.c

```diff
@@ -51,44 +78,57 @@
  *
  * @return register_value  - Value of the register.
 *******************************************************************************/
-uint8_t adt7420_get_register_value(struct adt7420_dev *dev,
-				   uint8_t register_address)
+uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
+				    uint8_t register_address)
 {
-	uint8_t register_value = 0;
-
-	i2c_write(dev->i2c_desc,
-		  &register_address,
-		  1,
-		  0);
-	i2c_read(dev->i2c_desc,
-		 &register_value,
-		 1,
-		 1);
-
-	return register_value;
+	if (adt7420_is_spi(dev))
+		return adt7420_spi_get_register_value(dev, register_address);
+	else
+		return adt7420_i2c_get_register_value(dev, register_address);
 }
 
 /***************************************************************************//**
- * @brief Sets the value of a register.
+ * @brief Sets the value of a register SPI/I2C.
  *
  * @param dev              - The device structure.
  * @param register_address - Address of the register.
- * @param register_value   - Value of the register.
+ * @param num_data_bytes   - Number of bytes.
+ * @param data             - Data to be written in input register.
+ * @param mask		   - Bit Mask of the bit to be written
+ * @param value		   - Value of the bit
  *
- * @return None.
+ * @return SUCCESS in case of success, FAILURE otherwise.
 *******************************************************************************/
-void adt7420_set_register_value(struct adt7420_dev *dev,
-				uint8_t register_address,
-				uint8_t register_value)
+int32_t set_register_value(struct adt7420_dev *dev,
+			   uint8_t register_address,
+			   uint8_t num_data_bytes,
+			   uint8_t *data, uint8_t mask, uint8_t value)
```

Comment:
This function seems to be a read-modify-write, but it's not self-contained. Could it work better if it had only the device handler, register address, mask and value as parameters? And read the current value of the register inside, change it with the mask and value, then write it?

This way it will assume some things that are not readily apparent to the user, that for example data is at least 2 bytes long or that it already contains the current value of the register.

Reply:
@adrimbarean - There can be a use case where we don't really need the current value of a register during a set register operation. 

Example:
In a case where we want to write to the one of the set point registers, we may want to write the value as such without any masking being performed. In such a scenario, we don't need to retain the current value held by the register before writing to it.  So should we really do the get register operation within this function?

Reply:
@adrimbarean -  Can we proceed with the existing flow itself? Or do you suggest modifying it?